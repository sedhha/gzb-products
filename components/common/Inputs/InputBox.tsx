import styled from 'styled-components';
type Props = {
  outlineColor: string;
};

const InputBox = styled.input`
  outline: none;
  border: none;
  background-color: transparent;
  font-size: 1rem;
  padding: 0.5rem;
  border-bottom: 0.5px solid ${(props: Props) => props.outlineColor};
`;

export default styled(InputBox)`
  &:focus {
    border-bottom: 2px solid ${(props: Props) => props.outlineColor};
  }
`;
