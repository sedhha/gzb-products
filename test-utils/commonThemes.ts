const outlineColor = 'red';
const borderRadius = '1rem';
const borderBottom = '0.5px solid red';

export const commonStyledVariables = {
  outlineColor,
  borderRadius,
  borderBottom,
};
