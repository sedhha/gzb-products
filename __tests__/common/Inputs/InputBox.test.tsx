import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import InputBox from '@common/Inputs/InputBox';
import { commonStyledVariables } from '@constants/test-utils/commonThemes';

describe('Input Box Test Cases', () => {
  it('Input Box should pick Outline Color dynamically: ', () => {
    const tree = renderer
      .create(<InputBox outlineColor={commonStyledVariables.outlineColor} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule(
      'border-bottom',
      commonStyledVariables.borderBottom
    );
  });
});
