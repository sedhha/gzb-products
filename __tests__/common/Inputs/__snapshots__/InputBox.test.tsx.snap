// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Box Test Cases Input Box should pick Outline Color dynamically:  1`] = `
.c0 {
  outline: none;
  border: none;
  background-color: transparent;
  font-size: 1rem;
  padding: 0.5rem;
  border-bottom: 0.5px solid red;
}

.c1:focus {
  border-bottom: 2px solid red;
}

<input
  className="c0 c1"
/>
`;
